src/complex/DataStructure/Factory/DataArrayFactory.hpp

```diff
@@ -1,23 +1,45 @@
 #pragma once
 
+#include "complex/DataStructure/DataArray.hpp"
+#include "complex/DataStructure/DataStore.hpp"
+#include "complex/DataStructure/DataStructure.hpp"
+#include "complex/Utilities/Parsing/HDF5/H5.hpp"
+#include "complex/Utilities/Parsing/HDF5/H5DataStructureReader.hpp"
+#include "complex/Utilities/Parsing/HDF5/H5GroupReader.hpp"
 #include "complex/Utilities/Parsing/HDF5/H5IDataFactory.hpp"
 
+#include <memory>
+#include <numeric>
+#include <optional>
+
 namespace complex
 {
 namespace H5
 {
+template <typename T>
 class COMPLEX_EXPORT DataArrayFactory : public IDataFactory
 {
 public:
-  DataArrayFactory();
+  DataArrayFactory()
+  : IDataFactory()
+  {
+    DataStructure ds;
+    using DataArrayType = DataArray<T>;
+    using DataStoreType = DataStore<T>;
+    DataArrayType* temp = DataArrayType::template CreateWithStore<DataStoreType>(ds, "TEMP", {1ULL}, {1ULL});
+    m_TypeName = temp->getTypeName();
```

Comment:
Also, if you are creating temp data, consider putting it in a temp DataStructure. Otherwise, you have to worry about cleaning it up, name conflicts, etc. If another filter forgets to clean up its temp data, that shouldn't prevent you from creating yours. Using RAII to clean up all temp data under all error conditions or branching return statements is a lot easier than manually handling it.